cpp/include/cudf/detail/reshape.hpp

```diff
@@ -33,4 +33,22 @@ std::unique_ptr<table> tile(
   cudaStream_t stream                 = 0,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 }  // namespace detail
+
+/**
+ * @brief Converts primitive types and string columns to lists of bytes, mimics Spark's cast to
+ * binary type.
```

Comment:
We shouldn't need to make reference to an external language/libraries semantics in describing C++ functionality. The C++ docs should standalone. 
```suggestion
 * @brief Converts a column elements to lists of bytes
```

An example would be nice here.

Reply:
added